### PR TITLE
Add better support for Operations

### PR DESCRIPTION
**Pull request by mjgiarlo**
Refs #118, #161, #160.

Sets up operations for batch creation, work creation, depositor changes, and file attachments, and surfaces the operations in the UI.

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

**Comment by jcoyne** [comment]
Should this wait for 1.0.0?

File: app/jobs/attach_files_to_work_job.rb [hunk 1]
```diff
@@ -2,43 +2,60 @@
...
       actor = Hyrax::Actors::FileSetActor.new(file_set, user)
       actor.create_metadata(visibility: work.visibility)
-      attach_content(actor, uploaded_file.file)
+      attach_content!(actor, uploaded_file.file, log)
       actor.attach_file_to_work(work)
       actor.file_set.permissions_attributes = work.permissions.map(&:to_hash)
 
       uploaded_file.update(file_set_uri: file_set.uri)
     end
+    # TODO: Should this depend on whether child_log was a success?
```
Comment (jcoyne) [inline app/jobs/attach_files_to_work_job.rb]: When the child log is finalized, it checks to see if it's parent has any open children. If they are all finalized, it sets the parent as a final state too (success or failure).

Comment (mjgiarlo) [inline app/jobs/attach_files_to_work_job.rb]: K, will remove TODOs.

**Comment by mjgiarlo** [comment]
@jcoyne Yes, this should probably wait 'til 0.1.0 is cut.

**Comment by mjgiarlo** [comment]
Please do not merge this PR. See :arrow_up:

**Comment by jcoyne** [comment]
@mjgiarlo ready for rebase & removal of TODOs

**Comment by jeremyf** [comment]
@mjgiarlo rebase time

**Comment by mjgiarlo** [comment]
I'll get back to this some time. If someone else wants to pick it up in the meantime, have at it.

**Comment by mjgiarlo** [comment]
Let's leave this hanging out here until we're ready to build out the operations feature more.

**Comment by no-reply** [comment]
Closing this as stale. I think at this point a rebase and refactor is unadvisable.

**Pull request closed without merging**